### PR TITLE
chore(move): refactoring move code with latest evm changes

### PR DESCRIPTION
**Pull request by Caglankaan**

### CODE REVIEW

File: move/move-ibc/sources/ibc.move [hunk 1]
```diff
@@ -501,118 +440,102 @@ module ibc::ibc {
...
             verify_connection_state(
                 connection,
                 proof_height,
                 proof_ack,
-                counterparty_conn_id,
+                counterparty_connection_id,
                 expected_connection
             );
-        assert!(err == 0, err);
+        assert!(err == 0, E_INVALID_PROOF);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:508]: revert plis

File: move/move-ibc/sources/ibc.move [hunk 2]
```diff
@@ -803,265 +712,326 @@ module ibc::ibc {
...
+
+        let connection_id = channel::connection_id(&chan);
+
+        let client_id = ensure_connection_state(connection_id);
 
-        let expected_counterparty = channel::new_counterparty(port_id, channel_id);
         let expected_channel =
             channel::new(
-                CHAN_STATE_TRYOPEN,
+                CHAN_STATE_INIT,
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:784]: revert, this should TRYOPEN

File: move/move-ibc/sources/ibc.move [hunk 3]
```diff
@@ -803,265 +712,326 @@ module ibc::ibc {
...
-                *channel::chan_counterparty_port_id(&chan),
                 counterparty_channel_id,
                 expected_channel
             );
-        assert!(err == 0, err);
-
-        channel::set_state(&mut chan, CHAN_STATE_OPEN);
-        channel::set_version(&mut chan, counterparty_version);
-        channel::set_chan_counterparty_channel_id(&mut chan, counterparty_channel_id);
+        assert!(err == 0, E_INVALID_PROOF);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:799]:
```suggestion
        assert!(err == 0, err);
```

File: move/move-ibc/sources/ibc.move [hunk 4]
```diff
@@ -803,265 +712,326 @@ module ibc::ibc {
...
+        commit_channel(channel_id, chan);
     }
 
     // Sends a packet
     public fun send_packet(
         ibc_app: &signer,
-        source_port: address,
-        source_channel: String,
-        timeout_height: height::Height,
+        source_port: address, // TODO: Verify this
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:888]: remove comment this is correct

File: move/move-ibc/sources/ibc.move [hunk 5]
```diff
@@ -803,265 +712,326 @@ module ibc::ibc {
...
+        packets: vector<Packet>,
+        proof_height: u64,
+        proof: vector<u8>,
+        intent: bool,
+        acknowledgement: vector<u8>
+    ) acquires IBCStore {
+        let l = vector::length(&packets);
+        assert!(l > 0, E_NOT_ENOUGH_PACKETS);
+
+        let firstPacket = *vector::borrow(&packets, 0);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:948]:
```suggestion
        let first_packet = *vector::borrow(&packets, 0);
```

File: move/move-ibc/sources/ibc.move [hunk 6]
```diff
@@ -803,265 +712,326 @@ module ibc::ibc {
...
+
+            let current_timestamp = timestamp::now_seconds() * 1_000_000_000; // 1e9
+            if (packet::timeout_timestamp(&packet) != 0) {
+                assert!(
+                    current_timestamp < packet::timeout_timestamp(&packet),
+                    E_TIMESTAMP_TIMEOUT
+                );
+            };
+
+            let commitmentKey =
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1003]:
```suggestion
            let commitment_key =
```

File: move/move-ibc/sources/ibc.move [hunk 7]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
-
+        process_receive(
+            packets,
+            proof_height,
+            proof,
+            false,
+            acknowledgement
+        );
+
+        // let channel =
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1059]: remove these comments

File: move/move-ibc/sources/ibc.move [hunk 8]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
+        if (!table::contains(&store.commitments, commitment_key)) {
+            abort E_PACKET_NOT_RECEIVED
+        };
+        let commitment = table::borrow(&store.commitments, commitment_key);
         assert!(
-            packet::source_channel(&packet)
-                == channel::chan_counterparty_channel_id(&channel),
-            E_SOURCE_AND_COUNTERPARTY_CHANNEL_MISMATCH
+            *commitment
+                == x"0100000000000000000000000000000000000000000000000000000000000000",
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1183]:
```suggestion
                == COMMITMENT_MAGIC,
```

File: move/move-ibc/sources/ibc.move [hunk 9]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
-    }
+        event::emit(WriteAcknowledgement { packet, acknowledgement });
+    }
 
     public fun acknowledge_packet(
         ibc_app: &signer,
-        port_id: address,
-        packet: packet::Packet,
-        acknowledgement: vector<u8>,
+        port_id: address, // TODO: Verify this
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1212]: remove comment

File: move/move-ibc/sources/ibc.move [hunk 10]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
-        let source_channel_id = *packet::source_channel(&packet);
-
-        let port_id = address_to_string(port_id);
-        assert!(port_id == *packet::source_port(&packet), E_UNAUTHORIZED);
-
-        let destination_port_id = *packet::destination_port(&packet);
-        let destination_channel_id = *packet::destination_channel(&packet);
-
-        let channel = ensure_channel_state(source_port_id, source_channel_id);
+        let firstPacket = *vector::borrow(&packets, 0);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1222]:
```suggestion
        let first_packet = *vector::borrow(&packets, 0);
```

File: move/move-ibc/sources/ibc.move [hunk 11]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
-                    destination_port_id,
-                    destination_channel_id,
-                    packet::sequence(&packet)
-                ),
-                hash::sha2_256(acknowledgement)
+                commitment_key,
+                commitment::commit_acks(acknowledgements)
             );
 
         if (err != 0) {
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1251]:
```suggestion
        assert!(err == 0, err);
```

File: move/move-ibc/sources/ibc.move [hunk 12]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
 
-        event::emit(AcknowledgePacket { packet, acknowledgement });
+            i = i + 1;
+        }
     }
 
     public fun timeout_packet(
         ibc_app: &signer,
-        port_id: address,
+        port_id: address, // TODO: Verify this
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1282]: remove this comment

File: move/move-ibc/sources/ibc.move [hunk 13]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
+        if (ordering == CHAN_ORDERING_ORDERED) {
+            let err =
+                verify_commitment(
+                    client_id,
+                    proof_height,
+                    proof,
+                    commitment::next_sequence_recv_commitment_key(destination_channel),
+                    bcs::to_bytes(&next_sequence_recv)
+                );
+            assert!(err == 0, E_INVALID_PROOF);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1310]:
```suggestion
            assert!(err == 0, err);
```

File: move/move-ibc/sources/ibc.move [hunk 14]
```diff
@@ -1071,322 +1041,295 @@ module ibc::ibc {
...
+        } else if (ordering == CHAN_ORDERING_UNORDERED) {
+            let commitment_key =
+                commitment::batch_receipts_commitment_key(
+                    destination_channel, commitment::commit_packet(&packet)
+                );
+            let err =
+                verify_absent_commitment(client_id, proof_height, proof, commitment_key);
+            assert!(err == 0, E_INVALID_PROOF);
```
Comment (aeryz) [inline move/move-ibc/sources/ibc.move:1318]:
```suggestion
            assert!(err == 0, err);
```

File: move/move-ibc/sources/packet.move [hunk 15]
```diff
@@ -1,97 +1,52 @@
...
     use std::vector;
-    use std::option::{Self, Option};
-    use std::hash;
-    use std::bcs;
-
-    use ibc::height::{Self, Height};
-    use ibc::proto_utils;
+    use ibc::ethabi;
 
+    const COMMITMENT_NULL: vector<u8> = x"0000000000000000000000000000000000000000000000000000000000000000";
```
Comment (aeryz) [inline move/move-ibc/sources/packet.move:5]: remove

File: move/ucs01-relay/sources/relay.move [hunk 16]
```diff
@@ -156,36 +149,31 @@ module ucs01::ibc {
...
+        let prefix = make_denom_prefix(channel_id);
         starts_with(denom, prefix)
     }
 
-    public fun make_denom_prefix(port_id: String, channel_id: String): String {
-        let prefix = port_id;
-        string::append_utf8(&mut prefix, b"/");
-        string::append(&mut prefix, channel_id);
+    public fun make_denom_prefix(channel_id: u32): String {
+        let channel_id_bytes = bcs::to_bytes<u8>(&(channel_id as u8)); // TODO: Verify this. Is it u8 or u32?
```
Comment (aeryz) [inline move/ucs01-relay/sources/relay.move:158]: make sure this is correct

**Review by aeryz — APPROVED** [review]

File: move/move-ibc/sources/eth_abi.move
Comment (benluelo) [inline move/move-ibc/sources/eth_abi.move:1]: rename file to `ethabi`?

File: move/move-ibc/sources/eth_abi.move [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+        // Append the padding
+        vector::append(buf, padding);
+    }
+
+    public fun encode_address(buf: &mut vector<u8>, addr: address) {
+        let sender_bytes = bcs::to_bytes(&addr);
+        vector::append(buf, sender_bytes);
+    }
+
+    public fun encode_uint<T: copy + store + drop>(
```
Comment (benluelo) [inline move/move-ibc/sources/eth_abi.move:40]: this doesn't account for packing?

**Pull request merged**